src/behaviours/pod-logs/download-logs.test.tsx

```diff
@@ -0,0 +1,269 @@
+      beforeEach(() => {
+        const button = rendered.getByTestId("download-logs-dropdown");
+
+        act(() => button.click());
+      });
+
+      it("shows download visible logs menu item", () => {
+        expect(rendered.getByTestId("download-visible-logs")).toBeInTheDocument();
+      });
+
+      it("shows download all logs menu item", () => {
+        expect(rendered.getByTestId("download-all-logs")).toBeInTheDocument();
+      });
+
+      describe("when call for logs resolves with logs", () => {
+        beforeEach(() => {
+          callForLogsMock.mockResolvedValue("all-logs");
+        });
+
+        describe("when selected 'download visible logs'", () => {
+          beforeEach(() => {
+            const button = rendered.getByTestId("download-visible-logs");
+
+            button.click();
+          });
+
+          it("shows save dialog with proper attributes", () => {
+            expect(openSaveFileDialogMock).toHaveBeenCalledWith("dockerExporter.log", "some-logs", "text/plain");
+          });
+        });
+
+        describe("when selected 'download all logs'", () => {
+          beforeEach(async () => {
+            await act(async () => {
+              const button = rendered.getByTestId("download-all-logs");
+
+              button.click();
+            });
+          });
+
+          it("logs have been called with query", () => {
+            expect(callForLogsMock).toHaveBeenCalledWith(
+              { name: "dockerExporter", namespace: "default" },
+              { "previous": true, "timestamps": false },
+            );
+          });
+
+          it("shows save dialog with proper attributes", async () => {
+            expect(openSaveFileDialogMock).toHaveBeenCalledWith("dockerExporter.log", "all-logs", "text/plain");
+          });
+
+          it("doesn't block download dropdown for interaction after click", async () => {
+            expect(rendered.getByTestId("download-logs-dropdown")).not.toHaveAttribute("disabled");
+          });
+        });
+
+        describe("blocking user interaction after menu item click", () => {
+          it("block download dropdown for interaction when selected 'download all logs'", async () => {
+            const downloadMenuItem = rendered.getByTestId("download-all-logs");
+
+            act(() => downloadMenuItem.click());
+
+            await waitFor(() => {
+              expect(rendered.getByTestId("download-logs-dropdown")).toHaveAttribute("disabled");
+            });
+          });
+
+          it("doesn't block dropdown for interaction when selected 'download visible logs'", () => {
+            const downloadMenuItem = rendered.getByTestId("download-visible-logs");
+
+            act(() => downloadMenuItem.click());
+
+            expect(rendered.getByTestId("download-logs-dropdown")).not.toHaveAttribute("disabled");
+          });
+        });
+      });
+
+      describe("when call for logs resolves with no logs", () => {
```

Comment:
Why show he save dialog for "visible logs" (and allow creating/saving an empty file), but show no save dialog for "all logs" (and show no message about what happened). Both seem odd behaviours.

Reply:
Rearranging test cases in the upcoming PR.